File: 066.PlusOne/Plus.java
public class Plus {
    public int[] plusOne(int[] digits) {
        int up = 1;
        for (int i = digits.length-1; i >= 0; i--) {
            int val = digits[i] + up;
            digits[i] = val % 10;
            up = val / 10;
        }

        if (up > 0) {
            int[] newDigits = new int[digits.length+1];
            newDigits[0] = up;
            for (int i = 0; i < digits.length; i++) {
                newDigits[i+1] = digits[i];
            }
            return newDigits;
        }

        return digits;
    }
}

File: 066.PlusOne/plus.py
class Solution(object):
    def plusOne(self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """
        up = 1
        for i in range(len(digits)-1, -1, -1):
            val = digits[i] + up
            digits[i] = val % 10
            up = val / 10

        if up:
            newDigits = [up]
            newDigits.extend(digits)
            return newDigits
        return digits

if __name__ == '__main__':
    print Solution().plusOne([9,9,9,9])
